### PR TITLE
Port options to use Context instead of a raw GuildMessageReceivedEvent

### PR DESCRIPTION
**Pull request by Kodehawa**
This also adds support for mention prefix. The porting was necessary to make this work.
Might need more changes and is untested. I need to give this a proper test on all options :)

This also splits GuildOptions into manageable classes and updates styling to Java 15. Should be ok now. 

Tested most changes and it seems to work, so review is probably gonna be on "what could be improved". A lot of changes though ;-;

### CODE REVIEW

File: src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java [hunk 1]
```diff
@@ -86,11 +89,27 @@ public I18nContext getGuildLanguageContext() {
     }
 
     public List<User> getMentionedUsers() {
-        return getEvent().getMessage().getMentionedUsers();
+        final var mentionedUsers = getEvent().getMessage().getMentionedUsers();
+        if (isMentionPrefix) {
+            final var mutable = new LinkedList<>(mentionedUsers);
```
Comment (natanbc) [inline src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java:94]: List#subList()

File: src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java [hunk 2]
```diff
@@ -86,11 +89,27 @@ public I18nContext getGuildLanguageContext() {
...
+        }
+
+        return mentionedUsers;
     }
 
     public List<Member> getMentionedMembers() {
-        return getEvent().getMessage().getMentionedMembers();
+        final var mentionedMembers = getEvent().getMessage().getMentionedMembers();
+        if (isMentionPrefix) {
+            final var mutable = new LinkedList<>(mentionedMembers);
```
Comment (natanbc) [inline src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java:106]: List#subList

File: src/main/java/net/kodehawa/mantarobot/commands/custom/legacy/CustomMessage.java [hunk 3]
```diff
@@ -17,48 +17,31 @@
...
-            return splitArgs(message.getContentStripped(), 2)[1];
-        }
-
-        return splitArgs(message.getContentStripped().replace(prefix, "").trim(), 2)[1];
+        return message;
     }
 
     public List<Member> getMentionedUsers() {
+        if (mentionPrefix) {
+            var mutable = new LinkedList<>(mentionedUsers);
```
Comment (natanbc) [inline src/main/java/net/kodehawa/mantarobot/commands/custom/legacy/CustomMessage.java:41]: List#subList

**Review by natanbc — CHANGES_REQUESTED** [review]

File: src/main/java/net/kodehawa/mantarobot/commands/custom/legacy/CustomMessage.java (hunk 3 above)
Comment (Kodehawa) [reply src/main/java/net/kodehawa/mantarobot/commands/custom/legacy/CustomMessage.java:41]: done

File: src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java (hunk 1 above)
Comment (Kodehawa) [reply src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java:94]: done

File: src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java (hunk 2 above)
Comment (Kodehawa) [reply src/main/java/net/kodehawa/mantarobot/core/modules/commands/base/Context.java:104]: done

**Pull request merged**